CRM/Case/BAO/Case.php

```diff
@@ -1041,7 +1041,10 @@ public static function getCaseActivity($caseID, &$params, $contactID, $context =
 
     // define statuses which are handled like Completed status (others are assumed to be handled like Scheduled status)
     $compStatusValues = array();
-    $compStatusNames = array('Completed', 'Left Message', 'Cancelled', 'Unreachable', 'Not Required');
+    $compStatusNames = array_merge(
+      array('Completed', 'Left Message', 'Cancelled', 'Unreachable', 'Not Required'),
+      CRM_Activity_BAO_Activity::getStatusesByType(CRM_Activity_BAO_Activity::COMPLETED)
+    );
     foreach ($compStatusNames as $name) {
       $compStatusValues[] = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_status_id', $name);
     }
```

Comment:
I don't think this loop is necessary anymore. If you make the above suggested change, then this is equal to `array_keys($compStatusNames)` (see `getStatusesByType` return value). Probably a bit more code cleanup below would allow us to just use one variable called `$completedStatuses` instead of one for names and the other for values.